bot/exts/moderation/alts.py

```diff
@@ -0,0 +1,169 @@
+    """A cog used to track a user's alternative accounts across Discord."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    def error_text_from_error(error: ResponseCodeError) -> str:
+        """Format the error into a user-facing message."""
+        if resp_json := error.response_json:
+            errors = ", ".join(
+                resp_json.get("non_field_errors", []) +
+                resp_json.get("source", []) +
+                resp_json.get("detail", [])
+            )
+            if errors:
+                return gettext.ngettext("Error from site: ", "Errors from site: ", len(errors)) + errors
+            return str(error.response_json)
+        return error.response_text
+
+    async def alts_to_string(self, alts: list[dict[str: str | int | list[int]]]) -> list[str]:
+        """Convert a list of alts to a list of string representations."""
+        lines = []
+        guild = self.bot.get_guild(self.bot.guild_id)
+        for idx, alt in enumerate(alts):
+            alt_obj = await get_or_fetch_member(guild, alt["target"])
+            alt_name = str(alt_obj) if alt_obj else alt["target"]
+            created_at = discord_timestamp(alt["created_at"])
+            updated_at = discord_timestamp(alt["updated_at"])
+
+            edited = f" edited on {updated_at}\n" if "edited" in alt else "\n"
+            num_alts = len(alt["alts"])
+            lines.append(
+                f"**Association #{idx} - {alt_name}**\n"
+                f"<@{alt['target']}> - {alt['target']}\n"
+                f"Issued by: <@{alt['actor']}> ({alt['actor']}) on {created_at}{edited}"
+                f"Context: {alt['context']}\n"
+                f"<@{alt['target']}> has {num_alts} associated {gettext.ngettext('account', 'accounts', num_alts)}"
+            )
+        return lines
+
+    @commands.group(name="association ", aliases=("alt", "alts", "assoc"), invoke_without_command=True)
+    async def association_group(
+        self,
+        ctx: commands.Context,
+        user_1: UnambiguousMemberOrUser,
+        user_2: UnambiguousMemberOrUser,
+        *,
+        context: str,
+    ) -> None:
+        """
+        Alternate accounts commands.
+
+        When called directly marks the two users given as alt accounts.
+        The context as to why they are believed to be alt accounts must be given.
+        """
+        try:
+            await self.bot.api_client.post(
+                f"bot/users/{user_1.id}/alts",
+                json={"target": user_2.id, "actor": ctx.author.id, "context": context},
+            )
+        except ResponseCodeError as e:
+            error = self.error_text_from_error(e)
+            await ctx.send(f":x: {error}")
+            return
+        await ctx.send(f"✅ {user_1.mention} and {user_2.mention} successfully marked as alts.")
+
+    @association_group.command(name="edit ", aliases=("e",))
+    async def edit_association_command(
+        self,
```

Comment:
Two things I've noticed here
1. When I use the `edit` alias, it tried to convert `edit` to an `UnambiguousMemberOrUser`

2. When you pass alts that don't exist yet, the error message isn't quite "pretty"
![image](https://github.com/python-discord/bot/assets/48383734/5896089a-95a3-4b20-83f0-d1801ae4f3ea)

Reply:
1. There was a space after `edit` in the name, so that's why it failed. I only ever tested with the `e` alias. Fixed now.
2. Fixed this too